### ISSUE TITLE
Fine-tuned Qwen2.5-Instruct isn't supported as expectation

### ISSUE THREAD

**Issue by sunday-hao** [ollama/ollama#8132]
### What is the issue?

Hi, there!
The base model I use is Qwen2.5-Instruct. Firstly, I fine-tuned the base model on my private dataset. Then, I convert it to .gguf format by llama.cpp. Trying to make my model easily supported by ollama.
but it turns out to be:
> ollama create my_model -f ./Modelfile 
> transferring model data 100%
> converting model
> Error: unsupported architecture

Also, I tried with `--quantize` and without `--quantize` argument, both failed with same error as above.
Nevertheless, the qwen2.5 series is supported by newest version of ollama based on [ollama.com/library](url). So is there anyone can help me? thanks a lot

### OS

Windows

### GPU

AMD

### CPU

AMD

### Ollama version

0.5.1

**Comment by sunday-hao** [comment]
And the command of llama.cpp used to convert the format of my fine-tuned model as below:
`python convert-hf-to-gguf.py --outfile path/to/outfile  path/to/safetensors`

contents of Modelfile as below:
`FROM /path/to/gguf_file`

**Comment by sunday-hao** [comment]
aaa

**Comment by rick-github** [comment]
Which fine tuner?

**Comment by sunday-hao** [comment]
Fullly fine-tune the base model with the tool of LLaMA-Factory

**Comment by sunday-hao** [comment]
And there is total 4 safetensors, like model-00001-of-00004.safetensors, model-00002-of-00004.safetensors, under the same root path that I used in the command when I was trying to convert the format to gguf by llama.cpp. Every safetensors is a part of the fine-tuned model, all of these safetensors together is the whole model.
And the memory of .gguf file is equal to the sum of the four .safetensors files approximately.

**Comment by sunday-hao** [comment]
PTAL when you have a chance. Just to make sure you are here. 
And thanks for your self-assignment.

**Comment by pdevine** [comment]
Sorry, I meant to comment earlier. The convert logic for Qwen hasn't been added to `ollama create` yet, but you can still convert the model to GGUF w/ llama.cpp's `convert_hf_to_gguf.py` script, and then import the resulting GGUF into ollama (again, using `ollama create`).

I'm not sure when I'll get time to add the convert logic directly, but I am working on some changes for `ollama create`.

**Comment by sunday-hao** [comment]
you mean `ollama create` is not available at present upon Qwen2.5 series both .safetensors and GGUF format. Is it right?

**Comment by pdevine** [comment]
@sunday-hao `ollama create` can pull in the GGUF model, but not the Safetensors model right now for Qwen 2.5. I'll add the logic for safetensors, but just haven't had time yet.

**Comment by sunday-hao** [comment]
if my Modelfile that points to the GGUF model under /home/Modelfile and my GGUF model converted from qwen2.5 under /home/downloads/.
So, the command is like `ollama create example -f /home/Modelfile`.
But, the error occured, unsupported arthitecture.
You mean the convert logic don't support qwen2.5 at present, but the convert logic supports other models?

**Comment by pdevine** [comment]
Make a Modelfile and include a line which looks like:

```
FROM /path/to/my/gguf/file.gguf
```

You can then call `ollama create -f Modelfile my-model`. This is assuming that the GGUF file was created correctly. The vision models are also not (yet) supported.

**Comment by sunday-hao** [comment]
yeah, I followed the pipeline strictly. And the model I convert is Qwen2.5-Instruct, but the error occurred.
According to your response, the reason is qwen2.5 is not supported at present.

**Comment by loodydo** [comment]
I am having the same problem when using Unsloth to fine tune and then attempt to import into Ollama. I essentially followed this guide:

`https://colab.research.google.com/drive/1WZDi7APtQ9VsvOrQSSC5DDtxq159j8iZ`

except I used Qwen2.5-Coder in place of llama-3.

After converting to GGUF and attempting to import to Ollama

Modelfile
`FROM C:\path_to\fine_tuned_qwen2.5_coder.gguf`

`ollama create qwen2.5-coder:7b-finetune -f Modelfile`

I get

converting model
Error: unsupported architecture

**Comment by rick-github** [comment]
The ollama converter doesn't support qwen2 at the moment.  Use [llama.cpp](https://github.com/ggerganov/llama.cpp/blob/master/convert_hf_to_gguf.py) to convert to GGUF and then import.

**Comment by loodydo** [comment]
@rick-github I must not understand what GGUF is very well. I assumed converting to GGUF would mean it was GGUF.

I used:
`model.save_pretrained_gguf("model", tokenizer, quantization_method = "q4_k_m")`
which completes without errors and creates "fine_tuned_qwen2.5_coder.gguf" which I then specify in my Modelfile.

I must be missing something fundamental.

**Comment by pdevine** [comment]
@loodydo it's just a file format. My guess is whatever library you're using to convert it filled in the `general.architecture` field incorrectly inside of the GGUF.

You can use the `convert_hf_to_gguf.py` script which you can find in `llm/llama.cpp/convert_hf_to_gguf.py`.

**Comment by loodydo** [comment]
@pdevine The problem may very well be in the gguf creation process.  I repeated the exact same fine tuning process were the only change was using a llama3.2 model instead of qwen2 and it imported into Ollama just fine. I will take a closer look at the gguf creation process and see what I find.

**Comment by pdevine** [comment]
@sunday-hao `ollama create` can't yet directly convert qwen 2.5 safetensors models right now, but the workaround is to use the `convert_hf_to_gguf.py` script that I mentioned.

**Comment by sunday-hao** [comment]
@loodydo Thanks for your try. I will convert it again.
